File: solutions/go/isogram/1/isogram.go
package isogram

import "strings"

func IsIsogram(word string) bool {
	isogram := true
    word = strings.ToUpper(word)
    for i, l := range word {
        if i+1 < len(word) && l != '-' && l != ' ' && strings.Contains(word[i+1:], string(l)) {
            isogram = false
            break;
        }
    }
    return isogram
}

File: solutions/csharp/isogram/1/Isogram.cs
public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var letters = new HashSet<char>();
        foreach (var letter in word.ToLower())
        {
            if (char.IsLetter(letter) && !letters.Add(letter))
                return false;
        }
        return true;
    }
}
